### PR TITLE
PYIC-5829: Changed COI routing to integrate with core-back journeys

### PR DESCRIPTION
**Pull request by RogVanGDS**
### What changed

Changed COI routing to integrate with core-back journeys

### Why did it change

Needed to tweak the journey events sent from the frontend to the backend as part of the coi journeys to distinguish between given names change and family name change 

### Issue tracking

- [PYIC-5829](https://govukverify.atlassian.net/browse/PYIC-5829)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-5829]: https://govukverify.atlassian.net/browse/PYIC-5829?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/ipv/middleware.test.js [hunk 1]
```diff
@@ -1298,7 +1298,7 @@ describe("journey middleware", () => {
     });
 
     it("should set journey to UPDATE_NAMES_DOB if detailsToUpdate is givenNames and lastName", async function () {
```
Comment (sam803) [inline src/app/ipv/middleware.test.js:1300]: We should probably update these test descriptions to match the changes made, or make them more generic

File: src/constants/update-details-journeys.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 module.exports = Object.freeze({
-  UPDATE_ADDRESS: "address",
+  UPDATE_ADDRESS: "address-only",
   UPDATE_NAMES_DOB: "names-dob",
-  UPDATE_NAME: "name",
-  UPDATE_NAME_ADDRESS: "name-address",
+  UPDATE_GIVEN_NAMES: "given-names-only",
+  UPDATE_FAMILY_NAME: "family-name-only",
+  UPDATE_GIVEN_NAME_ADDRESS: "given-names-and-address",
+  UPDATE_FAMILY_NAMES_ADDRESS: "family-name-and-address",
```
Comment (Wynndow) [inline src/constants/update-details-journeys.js]: The middleware is unhappy as it's trying to access these values using the below, which I think is right. So might need an update here.

```suggestion
  UPDATE_GIVEN_NAMES_ADDRESS: "given-names-and-address",
  UPDATE_FAMILY_NAME_ADDRESS: "family-name-and-address",
```

File: src/app/ipv/middleware.js [hunk 3]
```diff
@@ -402,8 +411,6 @@ module.exports = {
       PAGES.NO_PHOTO_ID_SECURITY_QUESTIONS_FIND_ANOTHER_WAY,
       PAGES.PAGE_MULTIPLE_DOC_CHECK,
       PAGES.PYI_CRI_ESCAPE,
-      PAGES.PYI_ESCAPE_M2B,
-      PAGES.PYI_KBV_ESCAPE_M2B,
```
Comment (Wynndow) [inline src/app/ipv/middleware.js:406]: I'm guessing this is an unrelated change - is it intentional?

Comment (RogVanGDS) [reply src/app/ipv/middleware.js:406]: Yeah they aren't being used anymore, the pages don't exist so I think we just forgot to remove them

**Review by Wynndow — APPROVED** [review]
